test/matcher.cpp

```diff
@@ -40,70 +40,183 @@ class MatcherMockMatchable : public Matchable {
 {
-	MatcherMockMatchable mock({{"abcd", "xyz"}});
 	Matcher m;
 
-	m.parse("abcd != \"uiop\"");
-	REQUIRE(m.matches(&mock));
+	SECTION("Works with strings") {
+		MatcherMockMatchable mock({{"abcd", "xyz"}});
+
+		REQUIRE(m.parse("abcd != \"uiop\""));
+		REQUIRE(m.matches(&mock));
+
+		REQUIRE(m.parse("abcd != \"xyz\""));
+		REQUIRE_FALSE(m.matches(&mock));
+	}
+
+	SECTION("Works with numbers") {
+		MatcherMockMatchable mock({{"answer", "42"}});
+
+		REQUIRE(m.parse("answer != 13"));
+		REQUIRE(m.matches(&mock));
 
-	m.parse("abcd != \"xyz\"");
-	REQUIRE_FALSE(m.matches(&mock));
+		REQUIRE(m.parse("answer != 42"));
+		REQUIRE_FALSE(m.matches(&mock));
+
+		SECTION("...but converts arguments to strings to compare") {
+			MatcherMockMatchable mock({{"agent", "007"}});
+
+			REQUIRE(m.parse("agent != 7"));
+			REQUIRE(m.matches(&mock));
+
+			REQUIRE(m.parse("agent != 007"));
+			REQUIRE_FALSE(m.matches(&mock));
+		}
+	}
+
+	SECTION("Doesn't work with ranges") {
+		MatcherMockMatchable mock({{"answer", "42"}});
+
+		REQUIRE(m.parse("answer != 0:100"));
+		REQUIRE(m.matches(&mock));
+
+		REQUIRE(m.parse("answer != 100:200"));
+		REQUIRE(m.matches(&mock));
+
+		REQUIRE(m.parse("answer != 42:200"));
+		REQUIRE(m.matches(&mock));
+
+		REQUIRE(m.parse("answer != 0:42"));
+		REQUIRE(m.matches(&mock));
+	}
 }
 
 TEST_CASE("Operator `=~` checks if field matches given regex", "[Matcher]")
 {
-	MatcherMockMatchable mock({{"AAAA", "12345"}});
 	Matcher m;
 
-	m.parse("AAAA =~ \".\"");
-	REQUIRE(m.matches(&mock));
+	SECTION("Works with strings") {
+		MatcherMockMatchable mock({{"AAAA", "12345"}});
 
-	m.parse("AAAA =~ \"123\"");
-	REQUIRE(m.matches(&mock));
+		REQUIRE(m.parse("AAAA =~ \".\""));
+		REQUIRE(m.matches(&mock));
 
-	m.parse("AAAA =~ \"234\"");
-	REQUIRE(m.matches(&mock));
+		REQUIRE(m.parse("AAAA =~ \"123\""));
+		REQUIRE(m.matches(&mock));
 
-	m.parse("AAAA =~ \"45\"");
-	REQUIRE(m.matches(&mock));
+		REQUIRE(m.parse("AAAA =~ \"234\""));
+		REQUIRE(m.matches(&mock));
 
-	m.parse("AAAA =~ \"^12345$\"");
-	REQUIRE(m.matches(&mock));
+		REQUIRE(m.parse("AAAA =~ \"45\""));
+		REQUIRE(m.matches(&mock));
+
+		REQUIRE(m.parse("AAAA =~ \"^12345$\""));
+		REQUIRE(m.matches(&mock));
+
+		REQUIRE(m.parse("AAAA =~ \"^123456$\""));
+		REQUIRE_FALSE(m.matches(&mock));
+	}
+
+	SECTION("Converts numbers to strings and uses them as regexes") {
+		MatcherMockMatchable mock({{"AAAA", "12345"}});
+
+		REQUIRE(m.parse("AAAA =~ 12345"));
+		REQUIRE(m.matches(&mock));
+
+		REQUIRE(m.parse("AAAA =~ 1"));
+		REQUIRE(m.matches(&mock));
+
+		REQUIRE(m.parse("AAAA =~ 45"));
+		REQUIRE(m.matches(&mock));
 
-	m.parse("AAAA =~ \"^123456$\"");
-	REQUIRE_FALSE(m.matches(&mock));
+		REQUIRE(m.parse("AAAA =~ 9"));
+		REQUIRE_FALSE(m.matches(&mock));
+	}
+
+	SECTION("Doesn't work with ranges") {
+		MatcherMockMatchable mock({{"AAAA", "12345"}});
+
+		REQUIRE(m.parse("AAAA =~ 0:123456"));
+		REQUIRE_FALSE(m.matches(&mock));
```

Comment:
Would this return true if the `mock` object was initialized with a value equal to `"0:123456"`?
e.g.: `MatcherMockMatchable mock({{"AAAA", "0:123456"}});`

Reply:
That's a great catch! It does return `true`, because it treats everything as strings :(

Well, not for long. The new parser is stricter.